libs/src/services/oauth/Oauth.ts

```diff
@@ -0,0 +1,254 @@
+import type { DiscoveryProvider, UserProfile } from '../discoveryProvider'
+
+export type LoginSuccessCallback = (profile: UserProfile) => void
+export type LoginErrorCallback = (errorMessage: string) => void
+export type ButtonOptions = {
+  size: 'small' | 'medium' | 'large'
+  corners: 'default' | 'pill'
+  customText: string
+  disableHoverGrow: boolean
+  fullWidth: boolean
+}
+
+// From https://stackoverflow.com/a/27747377
+const generateId = (): string => {
+  let arr = new Uint8Array(40 / 2) // Result of function will be 40 chars long
+  // @ts-ignore TS doesn't understand `msCrypto` (which provides compatibility for IE)
+  ;(window.crypto || window.msCrypto).getRandomValues(arr)
```

Comment:
We want to eventually be able to use the sdk on a server side node environment too. So we will need to infer the environment and use the node crypto library or `window.crypto` in the browser (and potentially allow this to be passed in as configuration, like for react-native). But we have a similar situation all throughout libs so we can address this later

Reply:
ahh yes gotcha. but a large part of this module also is opening the popup with `window.popup`, so i do think at least the `login` and `renderButton` utils here have to stay client side only

Reply:
💯